Add heatmap and discussion (#26)
greenelab/iscb-diversity-manuscript: content/20.results.md

We observed a multiclass area under the receiver operating characteristic curve (AUC) score of 95.4% for the classifier, indicating that the classifier can recapitulate name origins with high sensitivity and specificity.
For each individual region, the high AUC (above 94%, Fig. {@fig:wiki2019_lstm}A) suggests that our classifier was sufficient for use in a broad-scale examination of disparities.
We also observed that the model was well calibrated (Fig. {@fig:wiki2019_lstm}B).

> The confusion heatmap of log fold change of predicted over expected probability shows high multi-region classification accuracy compared to random chance (Fig. {@fig:wiki2019_lstm}C).

Reviewer: ```suggestion
We also examined potential systematic errors between pairs of nationality groupings with a confusion heatmap and did not find off-diagonal enrichment for any pairing (Fig. {@fig:wiki2019_lstm}C).
```

![The Wiki2019-LSTM model ranks the true nationality of Wikipedia names highly on testing data.
The area under the ROC curve is above 94% for each category, showing strong performance regardless of nationality (A).

> Calibration curve, computed with the caret R package, shows consistency between the predicted probabilities (midpoints of each fixed-width bin) and the observed fraction of names in each bin (B).

Reviewer: ```suggestion
A calibration curve, computed with the caret R package, shows consistency between the predicted probabilities (midpoints of each fixed-width bin) and the observed fraction of names in each bin (B).
```

> Log fold change probability confusion heatmap shows high multi-region classification accuracy compared to chance (C). Mean probability value in each predicted region is divided by the number of true names in that region (scaled to expected accuracy) to account for the imbalance of regions in the testing dataset.

Reviewer: ```suggestion
A log fold change probability confusion heatmap shows high multi-region classification accuracy compared to chance (C). The mean probability value in each predicted region is divided by the number of true names in that region (scaled to expected accuracy) to account for the imbalance of regions in the testing dataset.
```

> Log fold change probability confusion heatmap shows high multi-region classification accuracy compared to chance (C). Mean probability value in each predicted region is divided by the number of true names in that region (scaled to expected accuracy) to account for the imbalance of regions in the testing dataset.

Reviewer 2: ```suggestion
Heatmap showing whether names from a given region (x-axis) received higher (purple) or lower (green) predictions for each region (y-axis) than would be expected by region prevalence alone (C).
The values represent log~2~ fold change between the average predicted probability and the prevalence of the corresponding predicted region in the testing dataset (null).
Scaling by region prevalence accounts for the imbalance of regions in the testing dataset.
In all cases, the classifier predicts the true region above the expected null probability (matrix diagonal is all purple).
For off-diagonal cells, darker green indicates a lower mean prediction compared to the null.
For example, the classifier does not often mistake Hispanic names as Israeli, but is more prone to mistaking Muslim names as South Asian.
```
